crates/core/tedge_actors/src/builders.rs

```diff
@@ -79,81 +79,59 @@ pub trait RuntimeRequestSink {
 /// A trait to connect a message box under-construction to peer messages boxes
 pub trait ServiceProvider<Request: Message, Response: Message, Config> {
     /// Connect a peer message box to the message box under construction
-    fn connect_with(&mut self, peer: &mut impl ServiceConsumer<Request, Response>, config: Config);
+    fn connect_with(&mut self, peer: &mut impl ServiceConsumer<Request, Response, Config>);
 }
 
 /// A connection port to connect a message box under-connection to another box
-pub trait ServiceConsumer<Request: Message, Response: Message> {
+pub trait ServiceConsumer<Request: Message, Response: Message, Config> {
+    /// Return the config used by this actor to connect the service provider
+    fn get_config(&self) -> Config;
+
     /// Set the sender to be used by this actor's box to send requests
     fn set_request_sender(&mut self, request_sender: DynSender<Request>);
 
     /// Return a sender where the responses to this actor's box have to be sent
     fn get_response_sender(&self) -> DynSender<Response>;
 
     /// Connect this client message box to the service message box
-    fn connect_to<Config>(
-        &mut self,
-        service: &mut impl ServiceProvider<Request, Response, Config>,
-        config: Config,
-    ) where
+    fn connect_to(&mut self, service: &mut impl ServiceProvider<Request, Response, Config>)
+    where
         Self: Sized,
     {
-        service.connect_with(self, config)
+        service.connect_with(self)
     }
 
     /// Connect this client message box to the service message box
     ///
     /// Return the updated client message box.
-    fn connected_to<Config>(
-        mut self,
-        service: &mut impl ServiceProvider<Request, Response, Config>,
-        config: Config,
-    ) -> Self
+    fn connected_to(mut self, service: &mut impl ServiceProvider<Request, Response, Config>) -> Self
```

Comment:
> Not a big fan of this name connected_to as it doesn't really convey how it's different from connect_to without looking at the signature/doc.

Would `with_connection` help you to grasp from the name  that this creates a new enhanced builder?

Similarly, `connect_to` can be renamed `set_connection` to stress this mutates the builder.

> But, then I was wondering if we really need both connect_to and connected_to? How about just keeping this one that returns the Self? A user can choose to ignore the returned Self when he's not interested in chaining it any further.

This can work by taking a `&mut self` and returning the same `&mut self` once mutated.

Reply:
If we're sticking with both the methods, then I'd prefer the `with_xxx` variant as it gives that "builder" feel to it.